### PR TITLE
Let defaults be used in JSON

### PR DESCRIPTION
**Pull request by Womblue**
This PR adds a new option to `configure`, `--accept-defaults`. When used alongside `--answers`, this option enables profile to utilise the `default` and `smart_default` options for each question as backup answers if they aren't specified. Any required fields which don't have a default (including those which only have a smart default, and that smart default failed to produce an output) will still throw an error if not included.

### CODE REVIEW

File: lib/profile/commands/configure.rb [hunk 1]
```diff
@@ -44,26 +50,7 @@ def ask_questions
...
-                elsif (!question.validation.has_key?(:format) || output.match(Regexp.new(question.validation.format)))
-                  prefill ||= output
-                else
-                  smart_log.debug("Command result '#{output}' did not pass validation check for '#{question.text}'")
-                end
-              end
-              prefill ||= question.default
-              q.default prefill
-
+              q.default generate_prefill(question)
```
Comment (VoxSecundus) [inline lib/profile/commands/configure.rb:53]: `generate_prefill` doesn't exist under the `TTY::Prompt` scope; could you move this to just before the start of the `ask` block please.

Comment (Womblue) [reply lib/profile/commands/configure.rb:53]: I thought the functions still worked but instance variables didn't? I could've sworn I tested this and it worked, will check again

Comment (VoxSecundus) [reply lib/profile/commands/configure.rb:53]: The first time I ran `configure` it didn't work. I don't see why instance variables and instance methods would behave differently here.

Comment (Womblue) [reply lib/profile/commands/configure.rb:53]: You're right - annoyingly it needs to be outside of the entire `prompt.collect` block, which means that having it generate the prompt for the chosen question isn't so simple to do, will have more of a look and do some restructuring

**Review by VoxSecundus — APPROVED** [review]
This LGTM; if the test has worked for @ColonelPanicks I'm happy to merge.

**Pull request merged**